src/pocketmine/inventory/EnderChestInventory.php

```diff
@@ -0,0 +1,110 @@
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\inventory;
+
+use pocketmine\entity\Human;
+use pocketmine\level\Level;
+use pocketmine\network\mcpe\protocol\BlockEventPacket;
+use pocketmine\network\mcpe\protocol\LevelSoundEventPacket;
+use pocketmine\network\mcpe\protocol\types\WindowTypes;
+use pocketmine\Player;
+use pocketmine\tile\EnderChest;
+
+class EnderChestInventory extends ContainerInventory{
+
+	/** @var Human */
+	protected $owner;
+
+	/** @var FakeBlockMenu */
+	protected $holder;
+
+	public function __construct(Human $owner){
+		$this->owner = $owner;
+		parent::__construct(new FakeBlockMenu($this, $owner->getPosition()));
+	}
+
+	/**
+	 * @return Human
+	 */
+	public function getOwner(){
+		return $this->owner;
+	}
+
+	public function getNetworkType() : int{
+		return WindowTypes::CONTAINER;
+	}
+
+	public function getName() : string{
+		return "EnderChest";
+	}
+
+	public function getDefaultSize() : int{
+		return 27;
+	}
+
+	/**
+	 * Set the holders position to that of a tile
+	 *
+	 * @param EnderChest $enderChest
+	 */
+	public function openAt(EnderChest $enderChest) {
+		$this->holder->setComponents($enderChest->getX(), $enderChest->getY(), $enderChest->getZ());
+		$this->holder->setLevel($enderChest->level);
+	}
+
+	/**
+	 * This override is here for documentation and code completion purposes only.
+	 * @return FakeBlockMenu
+	 */
+	public function getHolder(){
+		return $this->holder;
+	}
+
+	public function onOpen(Player $who) : void{
+		parent::onOpen($who);
+
+		if(count($this->getViewers()) === 1 and ($level = $this->getHolder()->getLevel()) instanceof Level){
+			$this->broadcastBlockEventPacket(1, 2); //chest open
+			$level->broadcastLevelSoundEvent($this->getHolder()->add(0.5, 0.5, 0.5), LevelSoundEventPacket::SOUND_CHEST_OPEN);
+		}
+	}
+
+	public function onClose(Player $who) : void{
+		if(count($this->getViewers()) === 1 and ($level = $this->getHolder()->getLevel()) instanceof Level){
+			$this->broadcastBlockEventPacket(1, 0); //chest close
+			$level->broadcastLevelSoundEvent($this->getHolder()->add(0.5, 0.5, 0.5), LevelSoundEventPacket::SOUND_CHEST_CLOSED);
+		}
+		parent::onClose($who);
+	}
+
+	private function broadcastBlockEventPacket(int $case1, int $case2){
+		$pk = new BlockEventPacket();
+		$pk->x = $this->getHolder()->getX();
+		$pk->y = $this->getHolder()->getY();
+		$pk->z = $this->getHolder()->getZ();
+		$pk->case1 = $case1;
+		$pk->case2 = $case2;
+	}
```

Comment:
the code in here looks similar to that of ChestInventory, can it be reused instead of duplicating?